cglib/src/main/java/net/sf/cglib/core/Customizer2.java

```diff
@@ -0,0 +1,24 @@
+package net.sf.cglib.core;
+
+import org.objectweb.asm.Type;
+
+/**
+ * Customizes key types for {@link KeyFactory} right in constructor.
+ * This allows resolv
+ */
+public interface Customizer2 extends Customizer {
```

Comment:
This should be unified with Customizer. After all, the only difference should be the implementations built in KeyFactory